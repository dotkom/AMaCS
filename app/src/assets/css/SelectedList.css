.container {
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 8rem;
  padding: 0.5rem 0;
  background: #e0e0e0;
}

.committee {
  flex: 1;
  height: 100%;
  padding: 0 0.2rem;
  position: relative;
  padding: 0.75rem 0.5rem;
  display: flex;
  justify-content: center;
}

.committee + .committee {
  border-left: 1px solid #c4c4c4;
}

.number {
  position: absolute;
  top: -0.5rem;
  left: 0.25rem;
  font-size: 0.9rem;
  color: #636363;
}

@media(min-width: 600px) {
  .container {
    height: 9rem;
  }
}
