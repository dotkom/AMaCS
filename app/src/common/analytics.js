import ReactGA from 'react-ga'

const debug = process.env.NODE_ENV === 'development';
const trackingId = process.env.SG_GA_TRACKING_ID;

export function initializeAnalytics(history) {
  ReactGA.initialize(trackingId, {
    debug
  });
  history.listen(logPageView);
  logPageView(history.location);
}

export function logPageView(location) {
  if(trackingId) {
    ReactGA.set({ page: location.pathname + location.search });
    ReactGA.pageview(location.pathname + location.search);
  }
}
