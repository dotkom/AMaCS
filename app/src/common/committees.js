import arrkomIcon from '../../../static/arrkom-square.svg';
import bedkomIcon from '../../../static/bedkom-square.svg';
import dotkomIcon from '../../../static/dotkom-square.svg';
import fagkomIcon from '../../../static/fagkom-square.svg';
import prokomIcon from '../../../static/prokom-square.svg';
import trikomIcon from '../../../static/trikom-square.svg';

const committees = new Map([
  ['arrkom', {
    key: 'arrkom',
    name: "Arrkom",
    info: `Arrangementskomitéen jobber for at informatikkstudentene skal ha sosiale
    tilbud gjennom hele året. Vi arrangerer velkjente arrangement som
    immatrikuleringsball, blåtur, Åre-tur, kryssfest, filmkvelder og mye
    mer! Liker du at andre skal ha det kult? Er du en sosial og sprudlende
    person selv? Da er kanskje Arrangementskomitéen stedet for deg!`,
    icon: arrkomIcon,
  }],
  ['bedkom', {
    key: 'bedkom',
    name: "Bedkom",
    info: `Bedriftskomiteens oppgave er å knytte studentene og næringslivet
    sammen. Som medlem vil du være med å arrangere bedriftspresentasjoner
    hvor bedriftene får fortelle om arbeidsmetoder, miljø og prosjekter
    mm. På denne måten vil du opprette en tidlig kontakt med næringslivet
    og potensielle arbeidstakere under studietiden, samtidig som du får et
    innblikk i hvordan bedrifter rekrutterer studenter. Etter en
    bedriftspresentasjon tar gjerne bedriften studentene ut for å mingle
    og bli kjent under et godt måltid. Bedriftskomiteen er også
    knutepunktet til våre samarbeidspartnere i næringslivet ellers.`,
    icon: bedkomIcon,
  }],
  ['dotkom', {
    key: 'dotkom',
    name: "Dotkom",
    info: `Drifts- og utviklingskomiteen har ansvaret for alle linjeforeningens
    datasystemer. Vi leverer tjenester som nettside, intraweb og wiki,
    e-postsystemer, filområder og mye annet spennende. I en stor
    linjeforening er informasjonsflyt og kvalitet viktig. Drifts- og
    utviklingskomiteen prøver hele tiden å lage smarte løsninger som
    holder en høy standard, slik som i en bedrift. Er du interessert i
    programmering, drifting eller teknologi generelt? Vi kan tilby et
    unikt verv i en meget sosial gjeng som driver med IT på høyt nivå.
    Ingen forkunnskaper er påkrevd annet enn stor lærevilje!`,
    icon: dotkomIcon,
  }],
  ['fagkom', {
    key: "fagkom",
    name: "Fagkom",
    info: `Fag- og kurskomiteen tilbyr kunnskap om ny og spennende teknologi,
    gjennom blant annet kurs og foredrag. Vår jobb er å stimulere
    studentenes faglige interesser. Vi oppsøker kompetansen i
    næringslivet, og samarbeider tett med bedriftskomiteen for å komme i
    kontakt med de aktuelle bedriftene. Som medlem av Fag- og kurskomiteen
    vil du få god innsikt i emner utenfor skolen, og samtidig knytte
    tettere bånd til din fremtidige arbeidsgiver.`,
    icon: fagkomIcon,
  }],
  ['prokom', {
    key: "prokom",
    name: "Prokom",
    info: `Profil- og aviskomiteen sørger for at Onlines ansikt utad er
    profesjonelt og konsistent. Alt profileringsmateriale som klær,
    annonser, brev og brosjyrer, plakater, medaljer og andre effekter
    lages av oss. Vi har også en redaksjon som er ansvarlig for
    linjeforeningens tidsskrift, Offline. Har du en designerspire eller en
    skribent i magen? Vi leter etter deg som er interessert i design,
    markedsføring, journalisme eller skriving!`,
    icon: prokomIcon,
  }],
  ['trikom', {
    key: "trikom",
    name: "Trikom",
    info: `Trivselelskomiteen har ansvaret for trivsel blant informatikkstudentene.
    Dette inkluderer å arrangere små arrangementer i skoletiden,
    for eksempel vaffeldager eller andre ting som skaper trivsel.
    Vi er også ansvarlig for drift av kontoret som vil si å sette opp
    kontorvakter, møter, administrere tilganger og passe på at det er tilstrekkelig med mat, drikke, rekvisitter og lignende.
    Så hvis du har en gledesspreder i deg og liker å spre trivsel til dine medelever er trikom komiteen for deg.`,
    icon: trikomIcon,
  }],
]);

export default committees;
