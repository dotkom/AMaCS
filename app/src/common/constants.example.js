


export const OAUTH_SETTINGS = {
  authority: "http://localhost:8000",
  client_id: "...",
  response_type: "token",
  redirect_uri: "http://localhost:8080/auth",
  scope: "authentication.onlineuser.username.read authentication.onlineuser.first_name.read authentication.onlineuser.last_name.read authentication.onlineuser.email.read authentication.onlineuser.is_member.read authentication.onlineuser.field_of_study.read authentication.onlineuser.nickname.read authentication.onlineuser.rfid.read",
  metadata: {
    authorization_endpoint: "http://localhost:8000/sso/o/authorize/",
    userinfo_endpoint: "http://localhost:8000/sso/user/",
    end_session_endpoint: "http://localhost:8000/sso/o/revoke/"
  }
}
