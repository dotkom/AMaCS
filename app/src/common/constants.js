export const OAUTH_SETTINGS = {
  authority: process.env.SG_AUTH_AUTHORITY,
  client_id: process.env.SG_AUTH_CLIENT_ID,
  response_type: process.env.SG_AUTH_RESPONSE_TYPE,
  redirect_uri: process.env.SG_AUTH_REDIRECT_URI,
  scope: process.env.SG_AUTH_SCOPE,
  metadata: {
    authorization_endpoint: process.env.SG_AUTH_ENDPOINT_AUTH,
    userinfo_endpoint: process.env.SG_AUTH_ENDPOINT_USER,
    end_session_endpoint: process.env.SG_AUTH_ENDPOINT_REVOKE,
  }
}

export const API_SETTINGS = {
  host: process.env.SG_APPLICATION_BACKEND,
  application_endpoint: process.env.SG_APPLICATION_ENDPOINT
} 