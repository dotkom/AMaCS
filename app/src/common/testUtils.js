export function createCommitteeObject(
  name='Testkom',
  icon='MockIcon',
  info='Default info'
) {
  return {
    name, icon, info
  };
}
