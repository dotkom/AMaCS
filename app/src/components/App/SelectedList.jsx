import React from 'react';
import classNames from 'classnames';
import Selectable from "./Selectable"


import _s from 'assets/css/SelectedList.css';

function SelectedList({ committees, ordered, totalChoices, onChange }) {
  return (
    <div className={_s.container}>
      {
        [...Array(totalChoices)].map((_, i) => {
          const committee = (committees && committees.length > i)  ? committees[i] : null;
          const key = committee  ? committee.key : `empty${i}`;
          return (
            <div key={key} className={_s.committee}>
              { ordered && <div className={_s.number}>{ i + 1 }</div> }
              { committee && <Selectable onClick={() => onChange(committee.key)} small committee={committee} /> }
            </div>
          );
        })
      }
    </div>
  );
}

export default SelectedList;
