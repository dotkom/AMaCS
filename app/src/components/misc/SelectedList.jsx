import React from "react";
import { SortableContainer, SortableElement } from "react-sortable-hoc";
import Selectable from "./Selectable"

import _s from 'assets/css/SelectedList.scss';

const SortableSelectable = SortableElement(Selectable);

function SelectedList({ committees, disabled, totalChoices }) {
  return (
    <div className={_s.container}>
      {
        [...Array(totalChoices)].map((_, i) => {
          const committee = (committees && committees.length > i)  ? committees[i] : null;
          const key = committee  ? committee.key : `empty${i}`;
          return (
            <div key={key} className={_s.committee}>
              { !disabled && <div className={_s.number}>{ i + 1 }</div> }

              { committee && (
                <SortableSelectable
                  index={i}
                  committee={committee}
                  disabled={disabled}
                  draggable={!disabled} // Need this prop twice, as the HOC does not pass on disabled.
                  small
                />
              )}
            </div>
          );
        })
      }
    </div>
  );
}

export default SortableContainer(SelectedList);
