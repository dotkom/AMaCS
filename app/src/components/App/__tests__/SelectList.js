import React from 'react';
import { shallow } from 'enzyme';

import SelectedList from '../SelectedList';

import { createCommitteeObject } from 'common/testUtils';

describe('SelectedList', () => {
  it('renders correctly with no choices', () => {
    const wrapper = shallow(
      <SelectedList totalChoices={3} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with empty choices', () => {
    const wrapper = shallow(
      <SelectedList totalChoices={3} committees={[]} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with one out of three choices', () => {
    const committees = [
      createCommitteeObject('Testkom')
    ]
    const wrapper = shallow(
      <SelectedList totalChoices={3} committees={committees} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with two out of three choices', () => {
    const committees = [
      createCommitteeObject('Testkom'),
      createCommitteeObject('Bestkom'),
    ]
    const wrapper = shallow(
      <SelectedList totalChoices={3} committees={committees} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with three out of three choices', () => {
    const committees = [
      createCommitteeObject('Testkom'),
      createCommitteeObject('Bestkom'),
      createCommitteeObject('Sickkom'),
    ]
    const wrapper = shallow(
      <SelectedList totalChoices={3} committees={committees} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with more than total choices', () => {
    const committees = [
      createCommitteeObject('Testkom'),
      createCommitteeObject('Bestkom'),
      createCommitteeObject('Sickkom'),
    ]
    const wrapper = shallow(
      <SelectedList totalChoices={2} committees={committees} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onChange when clicking on a choice', () => {
    const committees = [
      createCommitteeObject('Testkom'),
      createCommitteeObject('Bestkom'),
      createCommitteeObject('Sickkom'),
    ]
    const onChangeMock = jest.fn();
    const wrapper = shallow(
      <SelectedList totalChoices={3} committees={committees} onChange={onChangeMock} />
    );
    wrapper.find('Selectable').first().simulate('click');
    expect(onChangeMock).toHaveBeenCalledWith('testkom');  });

  it('calls onChange when clicking on a choice', () => {
    const committees = [
      createCommitteeObject('Testkom'),
      createCommitteeObject('Bestkom'),
      createCommitteeObject('Sickkom'),
    ]
    const onChangeMock = jest.fn();
    const wrapper = shallow(
      <SelectedList totalChoices={3} committees={committees} onChange={onChangeMock} />
    );
    wrapper.find('Selectable').last().simulate('click');
    expect(onChangeMock).toHaveBeenCalledWith('sickkom');
  });

  it('renders correctly when ordered', () => {
    const committees = [
      createCommitteeObject('Testkom'),
    ];
    const wrapper = shallow(
      <SelectedList ordered totalChoices={3} committees={committees} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
