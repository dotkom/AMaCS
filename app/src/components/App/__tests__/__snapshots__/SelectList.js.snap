// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedList renders correctly when ordered 1`] = `
<div
  className="container"
>
  <div
    className="committee"
  >
    <div
      className="number"
    >
      1
    </div>
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "testkom",
          "name": "Testkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
  <div
    className="committee"
  >
    <div
      className="number"
    >
      2
    </div>
  </div>
  <div
    className="committee"
  >
    <div
      className="number"
    >
      3
    </div>
  </div>
</div>
`;

exports[`SelectedList renders correctly with empty choices 1`] = `
<div
  className="container"
>
  <div
    className="committee"
  />
  <div
    className="committee"
  />
  <div
    className="committee"
  />
</div>
`;

exports[`SelectedList renders correctly with no choices 1`] = `
<div
  className="container"
>
  <div
    className="committee"
  />
  <div
    className="committee"
  />
  <div
    className="committee"
  />
</div>
`;

exports[`SelectedList renders correctly with one out of three choices 1`] = `
<div
  className="container"
>
  <div
    className="committee"
  >
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "testkom",
          "name": "Testkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
  <div
    className="committee"
  />
  <div
    className="committee"
  />
</div>
`;

exports[`SelectedList renders correctly with three out of three choices 1`] = `
<div
  className="container"
>
  <div
    className="committee"
  >
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "testkom",
          "name": "Testkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
  <div
    className="committee"
  >
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "bestkom",
          "name": "Bestkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
  <div
    className="committee"
  >
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "sickkom",
          "name": "Sickkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
</div>
`;

exports[`SelectedList renders correctly with two out of three choices 1`] = `
<div
  className="container"
>
  <div
    className="committee"
  >
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "testkom",
          "name": "Testkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
  <div
    className="committee"
  >
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "bestkom",
          "name": "Bestkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
  <div
    className="committee"
  />
</div>
`;

exports[`SelectedList renders with more than total choices 1`] = `
<div
  className="container"
>
  <div
    className="committee"
  >
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "testkom",
          "name": "Testkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
  <div
    className="committee"
  >
    <Selectable
      committee={
        Object {
          "icon": "MockIcon",
          "info": "Default info",
          "key": "bestkom",
          "name": "Bestkom",
        }
      }
      onClick={[Function]}
      selected={false}
      small={true}
    />
  </div>
</div>
`;
